stacker/blueprints/base.py

```diff
@@ -402,6 +405,46 @@ def render_template(self):
         version = hashlib.md5(rendered).hexdigest()[:8]
         return (version, rendered)
 
+    def parse_user_data(self, raw_user_data):
+        """Translate a userdata file to into the file contents.
+
+        It supports referencing template variables to create userdata
+        that's supplemented with information from the data, as commonly
+        required when creating EC2 userdata files. Automatically, encodes
+        the data file to base64 after it is processed.
+
+        Args:
+            raw_user_data (str): the user data with the cloud-init info
+
+        Returns:
+            str: The parsed user data, with all the variables values and
+                 refs replaced with their resolved values.
+
+        """
+        pattern = re.compile(r'{{([::|\w]+)}}')
+        res = ""
+        start_index = 0
+        variables = self.get_variables()
+
+        for match in pattern.finditer(raw_user_data):
+            res += raw_user_data[start_index:match.start()]
+
+            key = match.group(1)
+
+            if key in variables:
+                if type(variables[key]) is CFNParameter:
+                    res += variables[key].to_parameter_value()
+                else:
+                    res += variables[key]
+            else:
+                raise MissingVariable(self.name, key)
+
+            start_index = match.end()
+
+        res += raw_user_data[start_index:]
+
+        return base64.b64encode(res)
```

Comment:
It might be better to not encode this at all, letting the user choose whether to encode it either using base64 or the Base64 troposphere method.  what do you think?